src/libexpr/eval-settings.cc

```diff
@@ -84,9 +84,14 @@ bool EvalSettings::isPseudoUrl(std::string_view s)
 
 std::string EvalSettings::resolvePseudoUrl(std::string_view url)
 {
-    if (hasPrefix(url, "channel:"))
+    if (hasPrefix(url, "channel:")) {
+        static bool haveWarned = false;
+        warnOnce(haveWarned,
+            "Channels are deprecated in favor of flakes in Determinate Nix. "
```

Comment:
```suggestion
            "`channel:` URLs are deprecated in favor of flakes in Determinate Nix. "
```

Can we also make it include the `channel:xxx` url they did use, and suggest updating it to point to `"https://nixos.org/channels/" + std::string(url.substr(8)) + "/nixexprs.tar.xz"`? _these_ URLs support the protocol we added to Nix for FlakeHub URLs.